internal/extractors/ramses_reader.go

```diff
@@ -18,53 +19,101 @@ type Packet struct {
-			header := ramses.Ramses{}
-			err = header.Read(stream.Buf)
-			if err != nil {
-				if err == io.EOF {
+		record := getRecord(stream)
+		if record.Error != nil {
+			recordChannel <- record
+		} else {
+			records = append(records, record)
+		}
+	}
+
+	sort.SliceStable(
+		records,
+		func(i int, j int) bool {
+			return records[i].RamsesHeader.Nanoseconds() < records[j].RamsesHeader.Nanoseconds()
+		},
+	)
+
+	for _, firstRecord := range records {
+		recordChannel <- firstRecord
+		for _, stream := range streamBatch {
+			if stream.Origin.Name != firstRecord.Origin.Name {
+				continue
+			}
+			for {
+				if stream.Buf.Len() == 0 {
+					break
+				}
+				record := getRecord(stream)
+				recordChannel <- record
+				if record.Error != nil {
 					break
 				}
-				recordChannel <- common.DataRecord{Origin: stream.Origin, Error: err, Buffer: []byte{}}
-				break
 			}
+		}
+	}
+}
 
-			if !header.Valid() {
-				err := fmt.Errorf("Not a valid RAC-file")
-				recordChannel <- common.DataRecord{Origin: stream.Origin, Error: err, Buffer: []byte{}}
-				break
-			}
+func getRecord(stream StreamBatch) common.DataRecord {
+	header := ramses.Ramses{}
+	err := header.Read(stream.Buf)
+	if err != nil {
+		return common.DataRecord{
+			Origin: stream.Origin,
+			Error:  fmt.Errorf("could not parse ramses header: %v (%v)", err, stream.Origin.Name),
+			Buffer: []byte{},
+		}
+	}
 
-			tmHeader := ramses.TMHeader{}
-			err = tmHeader.Read(stream.Buf)
-			if err != nil {
-				recordChannel <- common.DataRecord{Origin: stream.Origin, RamsesHeader: header, Error: err, Buffer: []byte{}}
-				break
-			}
-			header.Length -= uint16(binary.Size(tmHeader))
+	if !header.Valid() {
+		err := fmt.Errorf("Not a valid RAC-record %v (%s)", header, stream.Origin.Name)
+		return common.DataRecord{Origin: stream.Origin, Error: err, Buffer: []byte{}}
+	}
 
-			payload := make([]byte, header.Length)
-			_, err = stream.Buf.Read(payload)
-			if err != nil {
-				recordChannel <- common.DataRecord{Origin: stream.Origin, RamsesHeader: header, Error: err, Buffer: []byte{}}
-				break
-			}
-			recordChannel <- common.DataRecord{
-				Origin:         stream.Origin,
-				RamsesHeader:   header,
-				RamsesTMHeader: tmHeader,
-				Error:          nil,
-				Buffer:         payload,
-			}
+	tmHeader := ramses.TMHeader{}
+	err = tmHeader.Read(stream.Buf)
+	if err != nil && err != io.EOF {
+		return common.DataRecord{
+			Origin:       stream.Origin,
+			RamsesHeader: header,
+			Error:        fmt.Errorf("could not parse OHBSE CCDS TM Packet header: %v (%v)", err, stream.Origin.Name),
+			Buffer:       []byte{},
 		}
 	}
+	header.Length -= uint16(binary.Size(tmHeader))
+
+	payload := make([]byte, header.Length)
+	n, _ := stream.Buf.Read(payload)
+	if n != int(header.Length) {
+		return common.DataRecord{
+			Origin:         stream.Origin,
+			RamsesHeader:   header,
+			RamsesTMHeader: tmHeader,
+			Error: fmt.Errorf(
+				"payload truncaded, only found %v bytes but needed %v (%v)",
```

Comment:
You break differently here than on line 89 (I prefer this style).

Reply:
I was a matter of length of line. I have no breaks on 89 because it's shorter.